scripts/compress_images_test.py

```diff
@@ -0,0 +1,238 @@
+
+            if pathlib.Path(self.png_path).exists():
+                self.assertEqual(
+                    output_file_path.stat().st_size,
+                    pathlib.Path(self.png_path).stat().st_size
+                )
+
+    def test_compress_images_directory_removal_error(self) -> None:
+        """Test error handling when removing output directory fails."""
+        compressor = compress_images.ImageCompressor(
+            self.temp_dir,
+            output_dir=self.output_dir
+        )
+        os.makedirs(self.output_dir, exist_ok=True)
+        mock_rmtree = mock.MagicMock(ignore_errors=OSError('Permission denied'))
+        mock_log_error = mock.MagicMock()
+        mock_subprocess_run = mock.MagicMock(
+            return_value=mock.MagicMock(
+                returncode=1,
+                stderr='Permission denied error'
+            )
+        )
+
+        with self.swap(shutil, 'rmtree', mock_rmtree):
+            with self.swap(logging, 'error', mock_log_error):
+                with self.swap(subprocess, 'run', mock_subprocess_run):
+                    result_images: List[CompressedImageInfo] = [
+                        {
+                            'path': pathlib.Path(self.png_path),
+                            'original_size': 1000,
+                            'new_size': 500
+                        }
+                    ]
+
+                    compressor.compress_images(result_images)
+
+        mock_rmtree.assert_called_once_with(self.output_dir, ignore_errors=True)
+        mock_log_error.assert_called_with(
+            '[ERROR]: %s occurred on file %s',
+            pathlib.Path(self.png_path),
+            'Permission denied error'
+        )
+
+    def test_run_no_compressible_images(self) -> None:
+        """Test run method when no images are compressible."""
+        compressor = compress_images.ImageCompressor(self.temp_dir)
+        # Here we use object because a single compressible image
+        # dict needs to be wrapped in a list to match the
+        # method's expected return type and testing scenario.
+        with mock.patch.object(
+            compressor, 'find_compressible_images', return_value=[]
+        ):
+            result = compressor.run()
+            self.assertEqual(result, 0)
+
+    def test_run_with_compressible_images(self) -> None:
+        """Test run method with compressible images."""
+        input_dir = os.path.join(self.temp_dir, 'input')
+        output_dir = os.path.join(self.temp_dir, 'compressed_images')
+        os.makedirs(input_dir, exist_ok=True)
+
+        test_image_path = os.path.join(input_dir, 'test_image.png')
+        shutil.copy(self.png_path, test_image_path)
+
+        compressor = compress_images.ImageCompressor(
+            input_dir,
+            output_dir=output_dir
+        )
+        mock_compressible_images = [{
+            'path': pathlib.Path(test_image_path),
+            'original_size': 1000,
+            'new_size': 500
+        }]
```

Comment:
ditto: instead of using mocks to fake compressible images, why not use images that really are compressible in your test?

ditto elsewhere